### PR TITLE
[Network] New Commands for RC6

### LINKED ISSUE #846: [Network] Support new 2016-06-01 operations
With API version 2016-06-01, there are new operations that didn't exist before.
Add these operations to the CLI.

From a diff between the following files, I found the new operations listed below:
https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-network/2016-03-30/swagger/network.json
https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-network/2016-06-01/swagger/network.json

- [x] `nic get-effective-route-table`
- [x] `nic list-effective-nsg`
- [x] `vnet peering delete`
- [x] `vnet peering show`
- [x] `vnet peering create`
- [x] `vnet peering update` (generic update)
- [x] `vnet peering list`
- [x] `vnet check-ip-address`

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes #846. cc/ @DeepakRajendranMsft, @MikhailTryakhov for feedback.

Adds the following commands and tests:
- `nic get-effective-route-table`
- `nic list-effective-nsg`
- `vnet peering delete`
- `vnet peering show`
- `vnet peering create`
- `vnet peering update` (generic update)
- `vnet peering list`
- `vnet check-ip-address`

### CODE REVIEW

**Comment by tjprescott** [comment]
**nic show-effective-route-table**
```
Command
    az network nic show-effective-route-table: The get effective routetable operation retrieves all
    the route tables
    applied on a networkInterface.

Arguments

Resource Id Arguments
    --ids              : One or more resource IDs. If provided, no other 'Resource Id' arguments
                         should be specified.
    --name -n          : The network interface (NIC).
    --resource-group -g: Name of resource group.
```

**nic list-effective-nsg**
```
Command
    az network nic list-effective-nsg: The list effective network security group operation retrieves
    all the
    network security groups applied on a networkInterface.

Arguments

Resource Id Arguments
    --ids              : One or more resource IDs. If provided, no other 'Resource Id' arguments
                         should be specified.
    --name -n          : The network interface (NIC).
    --resource-group -g: Name of resource group.
```

**vnet peering create**
```
Command
    az network vnet peering create: Peerings in a VirtualNework resource.

Arguments
    --name -n           [Required]: The name of the VNET peering.
    --remote-vnet-id    [Required]: ID of the remote VNET.
    --resource-group -g [Required]: Name of resource group.
    --vnet-name         [Required]: The virtual network (VNET) name.
    --allow-forwarded-traffic     : Allows forwarded traffic from the VMs in the remote VNET.
    --allow-gateway-transit       : Allows gateway link to be used in the remote VNET.
    --allow-vnet-access           : Allows VMs in the remote VNET to access all VMs in the local
                                    VNET.
    --use-remote-gateways         : Allows VNET to use the remote VNET's gateway. Remote VNET
                                    gateway must have --allow-gateway-transit enabled for remote
                                    peering. Only 1 peering can have this flag enabled. Cannot be
                                    set if the VNET already has a gateway.
```

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py [hunk 1]
```diff
@@ -634,6 +634,31 @@ def body(self):
...
+        self.cmd('vm create -g {} -n {} --image UbuntuLTS --admin-password aBcD1234!@#$ --authentication-type password'.format(rg, vm))
+
+    def body(self):
+        rg = self.resource_group
+        vm = self.vm_name
+        nic_id = self.cmd('vm show -g {} -n {} --query "networkProfile.networkInterfaces[0].id"'.format(rg, vm))
+        self.cmd('network nic list-effective-nsg --ids {}'.format(nic_id),
+            checks=JMESPathCheck("length(value)", 1))
+        self.cmd('network nic show-effective-route-table --ids {}'.format(nic_id),
+            checks=JMESPathCheck("length(value)", 7))
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py]: Will there always be 7 or is this some special value?

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py [hunk 2]
```diff
@@ -827,13 +858,66 @@ def body(self):
         # Expecting the vnet we deleted to not be listed after delete
         self.cmd('network vnet list --resource-group {}'.format(self.resource_group), NoneCheck())
 
+class NetworkVNetPeeringScenarioTest(ResourceGroupVCRTestBase):
+    def __init__(self, test_method):
+        super(NetworkVNetPeeringScenarioTest, self).__init__(__file__, test_method, skip_teardown=True)
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py]: Was it intentional to leave `skip_teardown=True` in?
Just making sure :)

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py (hunk 1 above)
Comment (tjprescott) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py]: I couldn't get a JMESPath query to work to say "length(value) > 0". Any ideas?

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py (hunk 2 above)
Comment (tjprescott) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py]: Ack, NO that wasn't intentional!

**Comment by yugangw-msft** [comment]
Though English is not my first language, but the help text looks pretty awkward, particular `The list effective network security group operation retrieves...`, should be shrunk to `the operation retrieves`. Also the line break after `all the` is unnecessary.
I know this is caused by we are loading the SDK doc. I suggest we should override it, unless we have a plan to update across all.
```
az network nic list-effective-nsg: The list effective network security group operation retrieves
    all the
    network security groups applied on a networkInterface.
```

**Comment by yugangw-msft** [comment]
The rest LGTM

File: azure-cli.pyproj [hunk 3]
```diff
@@ -922,6 +922,17 @@
     <Content Include="command_modules\azure-cli-webapp\requirements.txt" />
   </ItemGroup>
   <ItemGroup>
+    <Interpreter Include="..\..\rc6\">
```
Comment (tjprescott) [inline azure-cli.pyproj]: Nope. I needed it while Derek's RC6 branch was still being worked on.

---
In reply to: [79086251](https://github.com/Azure/azure-cli/pull/903#discussion_r79086251) [](ancestors = 79086251)

**Comment by tjprescott** [comment]
Since we will need to improve help across the board, I went ahead and overrode the descriptions for the vnet and nic commands (not just the ones in this PR).

---
In reply to: [247488615](https://github.com/Azure/azure-cli/pull/903#issuecomment-247488615) [](ancestors = 247488615)

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py (hunk 1 above)
Comment (tjprescott) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py]: Actually I still never could get JMESPath to play nice here. I used a plain ol' assert.

---
In reply to: [79085537](https://github.com/Azure/azure-cli/pull/903#discussion_r79085537) [](ancestors = 79085537)

**Pull request merged**